torchvision/models/detection/roi_heads.py

```diff
@@ -477,8 +477,13 @@ def postprocess_detections(self, class_logits, box_regression, proposals, image_
         pred_scores = F.softmax(class_logits, -1)
 
         # split boxes and scores per image
-        pred_boxes = pred_boxes.split(boxes_per_image, 0)
-        pred_scores = pred_scores.split(boxes_per_image, 0)
+        if len(boxes_per_image) == 1:
+            # TODO : remove this when ONNX support dynamic split sizes
+            pred_boxes = (pred_boxes,)
```

Comment:
What is the limitation here exactly?

Reply:
I think that ONNX exporter doesn't yet support a `boxes_per_image` which is dynamic, so it gets converted into a constant and thus doesn't work for different number of boxes.

This is something that will be fixed by the ONNX team I believe, and this is currently only a workaround solution (which I think shouldn't affect torchscript?)

Reply:
exactly. split is not yet supported with dynamic sizes (so boxes_per_image will be exported as a constant which will for with any new input). But we are working on supporting this scenario in the exporter

Reply:
If the length isn't 1 isn't it still dynamic ? Not getting how this isn't still dynamic. 

(small change to support in TS, mostly just curious)

Reply:
it is, but we are currently only supporting a batch size of 1